src/memo/memo.ml

```diff
@@ -436,7 +436,8 @@ module Dep_node = M.Dep_node
 module Last_dep = M.Last_dep
 module Deps_so_far = M.Deps_so_far
 
-let no_deps_so_far : Deps_so_far.t = { set = Id.Set.empty; deps_reversed = [] }
+let no_deps_so_far () : Deps_so_far.t =
```

Comment:
Maybe call it `Deps_so_far.create`? If we're making the type mutable, probably good to use the corresponding naming idiom.

Reply:
Sounds good, done.